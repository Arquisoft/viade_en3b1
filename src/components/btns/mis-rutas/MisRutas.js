import React from 'react';
import Button from '@material-ui/core/Button';
import { MenuItem, Menu, Paper, Popper, Grow, MenuList, Typography, Link } from '@material-ui/core';
import ListIcon from '@material-ui/icons/List';
import EditLocationIcon from '@material-ui/icons/EditLocation';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { withStyles } from '@material-ui/core/styles';

const MisRutas = () => {

    const [anchorEl, setAnchorEl] = React.useState(null);

    const handleClick = (event) => {
        setAnchorEl(event.currentTarget);
    };

    const handleClose = () => {
        setAnchorEl(null);
    };

    return (
        <div>
            <Button
                aria-controls="customized-menu"
                aria-haspopup="true"
                variant="contained"
                color="primary"
                onClick={handleClick}
            >
                Activity
            </Button>
            <StyledMenu
                id="customized-menu"
                anchorEl={anchorEl}
                keepMounted
                open={Boolean(anchorEl)}
                onClose={handleClose}
            >
                <Link underline='none' href={'#/dashboard'}>
                    <StyledMenuItem>
                        <ListItemIcon>
                            <ListIcon />
                        </ListItemIcon>
                        <ListItemText primary="My routes" />
                    </StyledMenuItem>
                </Link>
                <Link underline='none' href={'#/create-route'}>
                    <StyledMenuItem>
                        <ListItemIcon>
                            <EditLocationIcon />
                        </ListItemIcon>
                        <ListItemText primary="Create route" />
                    </StyledMenuItem>
                </Link>
            </StyledMenu>
        </div>
    );
};

const StyledMenu = withStyles({
    paper: {
        border: '1px solid #d3d4d5',
    },
})((props) => (
    <Menu
        elevation={0}
        getContentAnchorEl={null}
        anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'center',
        }}
        transformOrigin={{
            vertical: 'top',
            horizontal: 'center',
        }}
        {...props}
    />
));

const StyledMenuItem = withStyles((theme) => ({
    root: {
        '&:focus': {
            backgroundColor: theme.palette.secondary.main,
        },
    },
}))(MenuItem);


export default MisRutas;