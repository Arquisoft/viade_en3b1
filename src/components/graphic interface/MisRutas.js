import React from 'react';
import Button from '@material-ui/core/Button';
import { MenuItem, Paper, Popper, Grow, MenuList, ListItemIcon, Typography } from '@material-ui/core';
import ListIcon from '@material-ui/icons/List';
import EditLocationIcon from '@material-ui/icons/EditLocation';

const MisRutas = () => {
    const [open, setOpen] = React.useState(false);
    const anchorRef = React.useRef(null);

    const handleToggle = () => {
        setOpen((prevOpen) => !prevOpen);
    };

    return (
        <div>
            <Button
                ref={anchorRef}
                aria-controls={open ? 'menu-list-grow' : undefined}
                aria-haspopup="true"
                onClick={handleToggle}
                color="inherit"
            >
                My Routes
            </Button>
            <Popper open={open} anchorEl={anchorRef.current} transition disablePortal>
                {({ TransitionProps, placement }) => (
                    <Grow
                        {...TransitionProps}
                        style={{ transformOrigin: placement === 'bottom' ? 'center top' : 'center bottom' }}
                    >
                        <Paper >
                            <MenuList >

                                <MenuItem onClick={() => window.location.href='#/myroutes'}>
                                    <ListItemIcon>
                                        <ListIcon />
                                    </ListItemIcon>
                                    <Typography>See routes</Typography>
                                </MenuItem>

                                <MenuItem onClick={() =>  window.location.href='#/newRoute'}>
                                    <ListItemIcon>
                                        <EditLocationIcon />
                                    </ListItemIcon>
                                    <Typography>Create new</Typography>
                                </MenuItem>

                            </MenuList>
                        </Paper>
                    </Grow>
                )}
            </Popper>
        </div>
    );
};

export default MisRutas;