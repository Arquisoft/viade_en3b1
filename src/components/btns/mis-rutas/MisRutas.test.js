import MisRutas from './MisRutas';

test('Define MisRutas', () => {
  expect(MisRutas).toBeDefined();
});

test('Renders correctly', () => {
  const wrapper = render(<MisRutas />);
  expect(wrapper).toMatchSnapshot();
});