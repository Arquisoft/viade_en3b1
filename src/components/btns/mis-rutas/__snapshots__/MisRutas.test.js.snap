// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <button
    aria-controls="customized-menu"
    aria-haspopup="true"
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-1 MuiButton-containedPrimary"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Your Activity 
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </span>
  </button>
</div>
`;
